atomicfu/src/concurrentTest/kotlin/kotlinx/atomicfu/locks/CyclicBarrierTest.kt

```diff
@@ -0,0 +1,148 @@
+package kotlinx.atomicfu.locks
+
+import kotlinx.atomicfu.atomic
+import kotlinx.atomicfu.atomicArrayOfNulls
+import kotlin.random.Random
+import kotlin.test.Test
+import kotlin.test.fail
+import kotlin.time.Duration
+
+class CyclicBarrierTest {
+    private class Arrs(numberOfThreads: Int) {
+        val after = atomicArrayOfNulls<Int>(numberOfThreads)
+        val before = atomicArrayOfNulls<Int>(numberOfThreads)
+        init {repeat(numberOfThreads) {
+            after[it].value = 0
+            before[it].value = 0
+        }}
+    }
+    @Test
+    fun simpleBarriertest() {
+        repeat(5) { iteration ->
+            println("Barrier test iteration $iteration")
+            (5..50 step 5).forEach { numberOfThreads ->
+                val barrier = CyclicBarrier(numberOfThreads)
+                val ar = Arrs(numberOfThreads)
+                val threads = List(numberOfThreads) { myThread ->
+                    Fut {
+                        repeat(numberOfThreads) { otherThread ->
+                            if (otherThread != myThread && ar.after[otherThread].value != 0) {
+                                fail("Thread $myThread arrived too early")
+                            }
+                        }
+                        sleepMills(Random.nextLong(100))
+                        ar.before[myThread].value = 1
+
+                        barrier.await()
+
+                        ar.after[myThread].value = 1
+
+                        repeat(numberOfThreads) { otherThread ->
+                            if (ar.before[otherThread].value == 0) {
+                                fail("Thread $myThread continued too early: $otherThread had value ${ar.before[otherThread].value}")
+                            }
+                        }
+                    }
+                }
+                Fut.waitAllAndThrow(threads)
+            }
+        }
+    }
+    
+    @Test
+    fun stressCyclicBarrier() {
+        repeat(5) { iteration ->
+            println("Stress test $iteration")
+            val threads = mutableListOf<Fut>()
+            val threadSetSize = (iteration + 1) * 5
+            val bar = CyclicBarrier(threadSetSize)
+            val syncBar = CyclicBarrier(threadSetSize * 5)
+            val ar = Arrs(threadSetSize * 5)
+            repeat(threadSetSize * 5) { tId ->
+                val t = Fut {
+                    repeat(50) { internalIteration ->
+                        sleepMills(Random.nextLong(100))
```

Comment:
In total, these `sleepMillis` affect test execution time considerably. Are they necessary?